pkg/discovery/discovery.go

```diff
@@ -51,35 +61,33 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) []error {
 		}
 	}
 
-	// closure used to collect and report errors.
-	rollup := func(err []error) {
-		if err != nil {
-			errlst = append(errlst, err...)
-		}
-	}
-
 	// 4. Run the plugin aggregator
-	errlst = append(errlst, pluginaggregation.Run(kubeClient, cfg.LoadedPlugins, cfg.Aggregation, outpath)...)
+	trackErrorsFor("running plugins")(
+		pluginaggregation.Run(kubeClient, cfg.LoadedPlugins, cfg.Aggregation, outpath),
+	)
 
 	// 5. Run the queries
-	rollup(QueryClusterResources(kubeClient, cfg))
+	trackErrorsFor("querying cluster resources")(
```

Comment:
Same logic as above, the querying of resources is only returning fatal errors now... non-fatal errors are being logged and not returned.